cli: Remove extra https in final output for app urls

briancain opened the pull request:
The deployUrl that comes in already will have https, so this removes the extra prepended https from the deployment url.

This matches how the Up CLI looks as well when it prints the deployment URL.

briancain committed:
cli: Remove extra https in final output for app urls

The deployUrl that comes in already will have https, so this removes the
extra prepended https from the deployment url.

briancain committed:
Add changelog

evanphx reviewed (changes requested):
I think if the url doesn't have a scheme on it, we should add https://. The DeployURL can come from plugins so they might not add a scheme.

briancain committed:
cli: Check for scheme on app URLs and append https if missing

This commit updates the various CLIs to look at the deploy and release
URLs to see if there is a scheme, and if not, include one prior to
printing the URLs.

briancain commented:
@evanphx I updated `pipeline run`, `up`, `deploy`, and `release` to append a scheme if missing via [d5fd63e](https://github.com/hashicorp/waypoint/pull/4208/commits/d5fd63e8b8fa1c1dcb334ccf12f5a7d52ccb9301)

briancain committed:
Update changelog

evanphx reviewed (approved):
We should probably DRY up that code eventually!

briancain commented:
Agreed - I opened an issue to track this for now! https://github.com/hashicorp/waypoint/issues/4209
